lib/logstash/pipeline.rb

```diff
@@ -321,4 +322,21 @@ def flush_filters_to_output!(options = {})
     end
   end # flush_filters_to_output!
 
+  private
+  def report_inflight_events(input_to_filter, filter_to_output, outputs)
+    loop do
+      sleep 5
+      report = []
+      report << "*** BEGIN INFLIGHT EVENTS REPORT ****"
+      report << "input_to_filter => #{input_to_filter.size} events" if input_to_filter.size > 0
+      report << "filter_to_output => #{filter_to_output.size} events" if filter_to_output.size > 0
+      outputs.each do |output|
+        next unless output.worker_queue
+        report << "#{output} => #{output.worker_queue.size} events" if output.worker_queue.size > 0
+      end
+      report << "*** END INFLIGHT EVENTS REPORT ****"
+      @logger.warn report.join("\n")
```

Comment:
This may make the report hard to read if users are logging to a file since `Object#inspect` (and in the future, json) is used to write to teh logstash logfile.